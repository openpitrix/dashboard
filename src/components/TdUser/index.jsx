import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import _ from 'lodash';

import styles from './index.scss';

const TdUser = ({ className, users, userId }) => {
  const user = _.find(users, { user_id: userId }) || {};

  return (
    <div className={classnames(styles.userShow, className)}>
      <label className={styles.name}>{user.username || userId}</label>
      {user.email}
    </div>
  );
};

TdUser.propTypes = {
  className: PropTypes.string,
  userId: PropTypes.string,
  users: PropTypes.array
};

TdUser.defaultProps = {
  users: [],
  userId: ''
};

export default TdUser;
