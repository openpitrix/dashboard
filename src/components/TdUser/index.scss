@import '~scss/vars';

.userShow {
  display: inline-block;
  font-size: 12px;
  color: $N75;

  .name {
    display: inline-block;
    margin-right: 4px;
    font-weight: 500;
    color: $N500;
  }
}
