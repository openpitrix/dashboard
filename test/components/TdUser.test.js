import TdUser from 'components/TdUser';

describe('Component TdUser', () => {
  const users = [
    {
      user_id: 'user-123',
      name: 'user123'
    },
    {
      user_id: 'user-133',
      name: 'user133'
    }
  ];

  it('basic render', () => {
    const wrapper = render(<TdUser users={users} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has find user', () => {
    const wrapper = render(<TdUser users={users} userId="user123" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
