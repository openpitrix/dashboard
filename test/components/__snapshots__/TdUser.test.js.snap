// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TdUser basic render 1`] = `
<div
  class="userShow"
>
  <label
    class="name"
  />
</div>
`;

exports[`Component TdUser has find user 1`] = `
<div
  class="userShow"
>
  <label
    class="name"
  >
    user123
  </label>
</div>
`;
